homeassistant/components/nmbs/sensor.py

```diff
@@ -294,3 +309,247 @@ def update(self):
+        """Return whether the connection goes through another station."""
+        if not self._attrs:
+            return False
+
+        return "vias" in self._attrs and int(self._attrs["vias"]["number"]) > 0
+
+    def update(self):
+        """Set the state to the duration of a connection."""
+        connections = self._api_client.get_connections(
+            self._station_from, self._station_to
+        )
+        self._next_trains = []
+
+        if connections is None or not connections["connection"]:
+            self._state = "No connections"
+        else:
+            for connection in connections["connection"]:
+                departure_delay = seconds_to_minutes(connection["departure"]["delay"])
+                arrival_delay = seconds_to_minutes(connection["arrival"]["delay"])
+                scheduled_departure_time_utc = dt_util.utc_from_timestamp(
+                    int(connection["departure"]["time"])
+                )
+                scheduled_arrival_time_utc = dt_util.utc_from_timestamp(
+                    int(connection["arrival"]["time"])
+                )
+                scheduled_departure_time_local = dt_util.as_local(
+                    scheduled_departure_time_utc
+                ).strftime("%H:%M")
+                scheduled_arrival_time_local = dt_util.as_local(
+                    scheduled_arrival_time_utc
+                ).strftime("%H:%M")
+                scheduled_departure_time_until = get_time_until(
+                    connection["departure"]["time"]
+                )
+                duration = get_ride_duration(
+                    connection["departure"]["time"],
+                    connection["arrival"]["time"],
+                    connection["departure"]["delay"],
+                    connection["arrival"]["delay"],
+                )
+                self._next_trains.append(
+                    {
+                        "departure_delay": departure_delay,
+                        "arrival_delay": arrival_delay,
+                        "duration": duration,
+                        "origin_name": connection["departure"]["station"],
+                        "destination_name": connection["arrival"]["station"],
+                        "departure_canceled": connection["departure"]["canceled"],
+                        "arrival_canceled": connection["arrival"]["canceled"],
+                        "left": connection["departure"]["left"],
+                        "leaves_in_minutes": scheduled_departure_time_until,
+                        "scheduled_departure_time": scheduled_departure_time_local,
+                        "scheduled_arrival_time": scheduled_arrival_time_local,
+                        "platform": connection["departure"]["platform"],
+                        "direction": connection["departure"]["direction"],
+                        "vehicle_id": connection["departure"]["vehicle"],
+                        ATTR_ATTRIBUTION: "https://api.irail.be/",
```

Comment:
The attribution should be extended, make a constant and then used by at all needed parts.